src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SecureGroovyScript.java

```diff
@@ -466,4 +456,14 @@ public FormValidation doCheckScript(@QueryParameter String value, @QueryParamete
 
     }
 
-}
+    @SuppressWarnings("unchecked")
+    public Descriptor<SecureGroovyScript> getDescriptor() {
```

Comment:
This is already implemented by the superclass, so I think you can delete it, right?

Reply:
First of all, thanks for checking my PR. I am using Jenkins and the Active Choice Plugin for years but Jenkins Pipeline only for one week. I got a NotSerializableException as I tried this:

`
import org.biouno.unochoice.model.GroovyScript
import org.jenkinsci.plugins.scriptsecurity.sandbox.groovy.SecureGroovyScript
import org.biouno.unochoice.CascadeChoiceParameter


node {  
     
    stage('SetParams') { 
        try {
    def sec = new GroovyScript(
        new SecureGroovyScript(
            """
             if ("first".equals(NormalChoice)) {
                 return ['yesFirst1', 'yesFirst2' ] 
             } else {
                 return ['notFirst1', 'notFirst2']
             }
            """, 
            true, 
            []
        ), 
        new SecureGroovyScript(
            "return ['Error']", 
            true, 
            []
        )  
    )
     def newParams = []
     newParams += [$class: 'ChoiceParameterDefinition', name: 'NormalChoice', choices: ['first', 'second']]
     newParams += new CascadeChoiceParameter('Param2', 'description', 'choice-parameter-5', sec, 'PT_SINGLE_SELECT', 'NormalChoice', false, 1)

     props = []
     props += [$class: 'ParametersDefinitionProperty', parameterDefinitions: newParams]
     properties(properties: props)
    } catch (Exception e) {
        println("Exception: ${e}")
    }
 
    }
    stage('Build') { 
       echo 'Building..'
    }
    stage('Deploy') { 
        echo 'Deploying....'
    }
}`

With my changes, I got rid of that exception. I will analyze all your comments and come back to you with questions (I guess).
Thanks!

Reply:
Hey,
I changed all the things you mentioned but after setting jenkins.version back to 2.60.3
I got a NoSuchFileException "hudson.Extension" in the target folder (with Intellij) and many test failures like:

> Caused by: com.gargoylesoftware.htmlunit.FailingHttpStatusCodeException: 500 Server Error for http://localhost:56777/jenkins/static/e6ca9e18/assets/jquery-detached/jsmodules/jquery2.js
        at com.gargoylesoftware.htmlunit.WebClient.throwFailingHttpStatusCodeExceptionIfNecessary(WebClient.java:590)
        at com.gargoylesoftware.htmlunit.html.HtmlPage.loadJavaScriptFromUrl(HtmlPage.java:1034)
        at com.gargoylesoftware.htmlunit.html.HtmlPage.loadExternalJavaScriptFile(HtmlPage.java:975)
        at com.gargoylesoftware.htmlunit.html.HtmlScript.executeScriptIfNeeded(HtmlScript.java:371)
        at com.gargoylesoftware.htmlunit.html.HtmlScript$2.execute(HtmlScript.java:246)
        at com.gargoylesoftware.htmlunit.javascript.JavaScriptEngine.doProcessPostponedActions(JavaScriptEngine.java:969)
        ... 54 more

> 
Do you know the cause?

Reply:
In what context do you get the exception, just from building the plugin? Do you get this when you run `mvn clean install -DskipTests` from the command line? Or is this when you run the plugin using `mvn hpi:run` or something?

Reply:
That said, your example usage seems strange to me. I think the idea I mentioned earlier would make more sense:

> Could [CascadeChoiceParameter] instead store the textual script as a String, [store the GroovyScript as a transient field], and then recreate the GroovyScript when deserialized?

Reply:
Hey,
with mvn hpi:run and mvn clean install -DskipTests I get these hudson.Extension file missing exception and when I manually add this file (got it when I build all with higher jenkins version) then I get (with  mvn hpi:run):

```
 WARNING: Failed to instantiate Key[type=org.jenkinsci.main.modules.sshd.SSHD, annotation=[none]]; skipping this component
com.google.inject.ProvisionException: Unable to provision, see the following errors:

1) Error injecting constructor, java.lang.NoClassDefFoundError: javax/xml/bind/DatatypeConverter
  at org.jenkinsci.main.modules.instance_identity.InstanceIdentity.<init>(InstanceIdentity.java:40)
  while locating org.jenkinsci.main.modules.instance_identity.InstanceIdentity
    for field at org.jenkinsci.main.modules.sshd.SSHD.identity(SSHD.java:35)

```
But maybe I don't need this fix anymore. But I think it is only like a workaround and
I can only imagine why the same pipeline code works (without the fix) when I write it
in another syntax:

```
def sec = [$class: 'GroovyScript',
        script: [$class: 'SecureGroovyScript', script: """ return ['yesFirst1', 'yesFirst2' ]  """, sandbox: true, classpath: null],
        fallbackScript: [$class: 'SecureGroovyScript', script: """ return ['ERROR'] """, sandbox: true, classpath: null]]
```
So this code works without my fix but (see above) code with "def sec = new GroovyScript ..." does not work (get NotSerializableException).

My question now is whether we should continue here. I think it would be nice (also for other developers) to continue. What do you think?
And your suggestion to save the script only as a string would of course be sufficient, but that would be a change in the Active Choice Jenkins plugin.

Reply:
> ```
> def sec = [$class: 'GroovyScript',
>         script: [$class: 'SecureGroovyScript', script: """ return ['yesFirst1', 'yesFirst2' ]  """, sandbox: true, classpath: null],
>         fallbackScript: [$class: 'SecureGroovyScript', script: """ return ['ERROR'] """, sandbox: true, classpath: null]]
> ```

That is the normal databinding syntax, although I'm surprised that it works without the fix. You could clean it up a bit using `@Symbol` on some of the types, but it's going to be awkward either way.

> My question now is whether we should continue here. I think it would be nice (also for other developers) to continue. What do you think?
> And your suggestion to save the script only as a string would of course be sufficient, but that would be a change in the Active Choice Jenkins plugin.

I think that it would be _best_ to change Active Choices Plugin so that it can be used from Pipeline more naturally, but the change here is harmless so I am fine with it. If you update the PR to just add `implements Serializable`, and remove the Jenkins update and formatting changes, we can see if the PR build passes and go from there.